### DISCUSSION TITLE
Filtering without query an API?

### DISCUSSION THREAD

**Discussion by ivanjeremic** [tbleckert/react-select-search#176] [General]
I have a normal Array of items and I want to make the filter work, in the example filtering is done by querying an API and applying an dynamic query key, how can I make the filter work on an normal array?

**Comment by tbleckert** [comment]
Hi and sorry for the late reply! You can use filterOptions prop for this. Like this:

```jsx
function filterOptions(options) {
    // We cache the options and return a function for the actual filtering
    return (q) => {
        return options.find((option) => option.name === q);
    };
}

<SelectSearch search filterOptions={filterOptions} .../>
```

Your filter function is called each time the options changes and your returned function is called on input change in the search field. The filtering happens on mount/options change as well, so you can use it to limit the amount of options to display. Check this example https://react-select-search.com/?path=/story/single-select--limited-options
